sc-poc/modules/auth/app.go

```diff
@@ -33,42 +23,44 @@ func (App) CaddyModule() caddy.ModuleInfo {
 	}
 }
 
-// Provision sets up the auth module.W
 func (a *App) Provision(ctx caddy.Context) error {
 	a.logger = ctx.Logger(a)
-
-	// Load the dependent apps
-	graphqlTemp, err := ctx.App("graphql")
+	sourceManT, err := ctx.App("source")
 	if err != nil {
-		a.logger.Error("Unable to load the graphql application", zap.Error(err))
-		return err
+		a.logger.Error("Unable to load the source manager", zap.Error(err))
 	}
-	a.graphqlApp = graphqlTemp.(*graphql.App)
+	sourceMan := sourceManT.(*source.App)
+
+	// Get all relevant sources
+	sources := sourceMan.GetSources("auth")
+	a.secrets = []SecretSource{}
+	a.policies = make(map[string]PolicySource)
+	for _, src := range sources {
+
+		// First resolve the source's dependencies
+		if err := source.ResolveDependencies(ctx, "auth", src); err != nil {
+			a.logger.Error("Unable to resolve source's dependency", zap.String("source", src.GetName()), zap.Error(err))
+			return err
+		}
 
-	// Compile the rego policies
-	if err := a.compileRegoPolicies(); err != nil {
-		a.logger.Error("Unable to compile rego policies", zap.Error(err))
-		return err
+		// Get sources with secret
+		s, ok := src.(SecretSource)
+		if ok {
+			a.secrets = append(a.secrets, s)
+		}
+
+		// Get sources with policy
+		p, ok := src.(PolicySource)
+		if ok {
+			a.policies[p.GetName()] = p
```

Comment:
The PolicySource object doesn't need a GetName since the `source.Source` already implements it